### ISSUE TITLE
Hale connect login error

### ISSUE THREAD

**Issue by apfelnymous** [halestudio/hale#702]
Hello, when I login to hale connect from the hale studio dialog the following error occurs:

<eu.esdihumboldt.hale.io.haleconnect.HaleConnectException: Failed to connect to haleconnect.com/35.157.201.132:443
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.login(HaleConnectServiceImpl.java:345)
	at eu.esdihumboldt.hale.io.haleconnect.ui.HaleConnectLoginHandler$1.run(HaleConnectLoginHandler.java:92)
	at eu.esdihumboldt.hale.ui.util.io.ThreadProgressMonitor$1.run(ThreadProgressMonitor.java:121)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.net.ConnectException: Failed to connect to haleconnect.com/35.157.201.132:443
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143)
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
	at com.squareup.okhttp.Call.getResponse(Call.java:286)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.execute(Call.java:80)
	at com.haleconnect.api.user.v1.ApiClient.execute(ApiClient.java:969)
	at com.haleconnect.api.user.v1.api.LoginApi.loginWithHttpInfo(LoginApi.java:141)
	at com.haleconnect.api.user.v1.api.LoginApi.login(LoginApi.java:127)
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.login(HaleConnectServiceImpl.java:324)
	... 3 more>

**Comment by florianesser** [comment]
Hi @apfelnymous, thanks for getting in touch. What version of hale studio are you using?

**Comment by apfelnymous** [comment]
Hey its 3.4.0 on windows 7 64, Ive been fine with that release on Ubuntu 18.04 yesterday.
Ill try the new release shortly and let you know.

**Comment by stempler** [comment]
Could it be that you need to use a proxy server to connect to the internet? That's often an issue as this has to be configured in hale studio explicitly (Menu Window -> Settings)

**Comment by apfelnymous** [comment]
Hey, an update: I tried the same on Hale 3.4.1, also a colleague filled the proxy fields. Still the same error.

Kind regards

**Comment by stempler** [comment]
If there is a general issue with the proxy configuration a test is also if the background map in the map view can be shown. Thus you could verify that the proxy settings are correct.

**Comment by florianesser** [comment]
@apfelnymous Is this still an issue for you? Were you able to verify your proxy settings by checking the Map view?

**Comment by apfelnymous** [comment]
Hello @florianesser , from coming week on Im employed to work and I can try this again.
Kind regards !

**Comment by apfelnymous** [comment]
Hey, I just checked the Map view, there is data shown, it appears to be working, we tried to fill in the proxy settings and I tried to login into hale connect again, didnt work, although my hale connect credentials are correct. Hale returned the following error:

" eu.esdihumboldt.hale.io.haleconnect.HaleConnectException: Failed to connect to haleconnect.com/52.29.224.56:443
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.login(HaleConnectServiceImpl.groovy:349)
	at eu.esdihumboldt.hale.io.haleconnect.ui.HaleConnectLoginHandler$1.run(HaleConnectLoginHandler.java:92)
	at eu.esdihumboldt.hale.ui.util.io.ThreadProgressMonitor$1.run(ThreadProgressMonitor.java:122)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.net.ConnectException: Failed to connect to haleconnect.com/52.29.224.56:443
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143)
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
	at com.squareup.okhttp.Call.getResponse(Call.java:286)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.execute(Call.java:80)
	at com.haleconnect.api.user.v1.ApiClient.execute(ApiClient.java:969)
	at com.haleconnect.api.user.v1.api.LoginApi.loginWithHttpInfo(LoginApi.java:141)
	at com.haleconnect.api.user.v1.api.LoginApi.login(LoginApi.java:127)
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.login(HaleConnectServiceImpl.groovy:328)
	... 3 more
"

Btw: When using the "validate credentials" function, it will also terminate with an error:

"eu.esdihumboldt.hale.io.haleconnect.HaleConnectException: java.net.ConnectException: Failed to connect to haleconnect.com/52.29.224.56:443
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.verifyCredentials(HaleConnectServiceImpl.groovy:489)
	at eu.esdihumboldt.hale.io.haleconnect.ui.preferences.HaleConnectPreferencePage$4.widgetSelected(HaleConnectPreferencePage.java:135)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4118)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1052)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3931)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3534)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at org.eclipse.ui.internal.dialogs.WorkbenchPreferenceDialog.open(WorkbenchPreferenceDialog.java:211)
	at org.eclipse.ui.internal.handlers.ShowPreferencePageHandler.execute(ShowPreferencePageHandler.java:59)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:291)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:93)
	at sun.reflect.GeneratedMethodAccessor489.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:318)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:252)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:161)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:204)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:433)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:446)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:472)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4118)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1052)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3931)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3534)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1170)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1059)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:667)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:597)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at eu.esdihumboldt.hale.ui.application.Application.run(Application.java:116)
	at eu.esdihumboldt.hale.ui.application.Application.run(Application.java:1)
	at eu.esdihumboldt.hale.common.app.AbstractApplication.run(AbstractApplication.java:51)
	at eu.esdihumboldt.hale.common.app.AbstractApplication.start(AbstractApplication.java:33)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:592)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1498)
Caused by: com.haleconnect.api.user.v1.ApiException: java.net.ConnectException: Failed to connect to haleconnect.com/52.29.224.56:443
	at com.haleconnect.api.user.v1.ApiClient.execute(ApiClient.java:973)
	at com.haleconnect.api.user.v1.api.LoginApi.loginWithHttpInfo(LoginApi.java:141)
	at com.haleconnect.api.user.v1.api.LoginApi.login(LoginApi.java:127)
	at eu.esdihumboldt.hale.io.haleconnect.internal.HaleConnectServiceImpl.verifyCredentials(HaleConnectServiceImpl.groovy:483)
	... 56 more
Caused by: java.net.ConnectException: Failed to connect to haleconnect.com/52.29.224.56:443
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143)
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
	at com.squareup.okhttp.Call.getResponse(Call.java:286)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.execute(Call.java:80)
	at com.haleconnect.api.user.v1.ApiClient.execute(ApiClient.java:969)
	... 59 more
"

**Comment by florianesser** [comment]
Hi @apfelnymous, were you able to resolve this problem in the end?

**Comment by apfelnymous** [comment]
Hi @florianesser , when I started using Hale Studio on a machine where no proxy credentials are needed - only the proxy host url, I stopped receiving the error and can login successfully. However when I search for alignments in the search dialogue, another error will return right after running the search function. When I confirm the error, the search results will reset.
![HaleConnectFehler](https://user-images.githubusercontent.com/7989160/82024264-cfdbce00-968f-11ea-960c-59e971f3ca4e.PNG)
